### PR TITLE
Check for depth coherence at the lowest level for YCgCo-R

### PR DESCRIPTION
**Pull request by vrabaud**
For now, the depth check is done at the level of PNG and JPEG reader.

### CODE REVIEW

File: src/reformat.c [hunk 1]
```diff
@@ -204,6 +204,16 @@ avifResult avifImageRGBToYUV(avifImage * image, const avifRGBImage * rgb)
         return AVIF_RESULT_NOT_IMPLEMENTED;
     }
 
+#if defined(AVIF_ENABLE_EXPERIMENTAL_YCGCO_R)
+    if (state.mode == AVIF_REFORMAT_MODE_YCGCO_RE || state.mode == AVIF_REFORMAT_MODE_YCGCO_RO) {
+        const int bitOffset = (state.mode == AVIF_REFORMAT_MODE_YCGCO_RE) ? 2 : 1;
+        const int maxValue = (1 << (state.yuvDepth - bitOffset)) - 1;
```
Comment (wantehchang) [inline src/reformat.c:210]: Nit/optional: Here we can compare `state.yuvDepth` or `image->depth` with `rgb->depth` directly and skip the calculation of `maxValue`, e.g.,

```
        if (state.yuvDepth - bitOffset != rgb->depth) {
            return AVIF_RESULT_UNSUPPORTED_DEPTH;
        }
```

File: src/reformat.c [hunk 2]
```diff
@@ -563,6 +572,19 @@ static avifResult avifImageYUVAnyToRGBAnySlow(const avifImage * image,
     const uint16_t yuvMaxChannel = (uint16_t)state->yuvMaxChannel;
     const float rgbMaxChannelF = state->rgbMaxChannelF;
 
+#if defined(AVIF_ENABLE_EXPERIMENTAL_YCGCO_R)
+    int maxValue;
+    if (state->mode == AVIF_REFORMAT_MODE_YCGCO_RE || state->mode == AVIF_REFORMAT_MODE_YCGCO_RO) {
+        const int bitOffset = (state->mode == AVIF_REFORMAT_MODE_YCGCO_RE) ? 2 : 1;
+        maxValue = (1 << (state->yuvDepth - bitOffset)) - 1;
```
Comment (wantehchang) [inline src/reformat.c:579]: Nit: The variable `maxValue` is now used in a much larger scope, so its name should indicate whether it's YUV or RGB.

However, I found that we can use `state->rgbMaxChannel` in the YCgCo-R code at lines 765-767, so here we can also just do a direct comparison of bit depths as I suggested in line 210.

**Review by wantehchang — COMMENTED** [review]
Thanks for the fix.

File: src/reformat.c (hunk 1 above)
Comment (vrabaud) [reply src/reformat.c]: Done

File: src/reformat.c (hunk 2 above)
Comment (vrabaud) [reply src/reformat.c]: Simpler indeed. So many members ... Done.

**Review by wantehchang — APPROVED** [review]
LGTM.

It just occurred to me that we may be able to check for bit depth coherence inside the `avifPrepareReformatState` function, which is called by both `avifImageRGBToYUV` and `avifImageYUVToRGB`. Would you like to try that before I merge this pull request?

**Comment by vrabaud** [comment]
Indeed, that is the right spot for a check.

**Review by wantehchang — APPROVED** [review]
LGTM. Thanks. I will merge the pull request for you.

**Pull request merged**